pyproject.toml

```diff
@@ -1,8 +1,9 @@
 [tool.poetry]
-name = "otar-biocypher"
+name = "open_targets"
 version = "0.3.0"
 description = "BioCypher Open Targets Knowledge Graph adapter"
 authors = ["Sebastian Lobentanzer <sebastian.lobentanzer@gmail.com>"]
+packages = [{include = "open_targets/"}]
```

Comment:
I think this is a building argument and does not do what you think it does because the lock file has no change related to it. I think the script just works without this and the original workaround. I tested the `main` branch with the workaround deleted and it worked. I don't why it didn't work but I also don't think this is fixing anything as well.

I think we better remove it until we are sure how we want to distribute the adapter.

Reply:
Agree; we are not building the package to distribute at the moment. While this could be an avenue, we haven't gone so far as to put any adapter on PyPI yet. If we create a package, I am thinking it would be more a central one that contains multiple standard adapters, not each one individually.

Reply:
@laulopezreal According to poetry doc, the default mode is already package-mode so the project is installed in editable mode and I can confirm that with branch `main` I see this in a newly created virtual environment containing this package manifest:
```json
{"dir_info": {"editable": true}, "url": "file:///C:/Users/Paul/Documents/GitHub/open-targets"}
```

Adding this line produces the same package manifest so I don't think your change here matters.

Reply:
@kpto, thanks for the explanation. Do we actually need an editable installation?

Reply:
@slobentanzer I think it depends how we want to distribute it and we don't need to make our decision now. I just want to point out that the change here is redundant and should be removed.

Reply:
While it is redundant, it is often preferred to be explicit. The configuration entry at least leaves no question of the install behaviour. If it really does not cause harm anywhere, I would still keep it, as it more explicitly states what the package install is doing.

I am again coming from the angle of the not-very-experienced programmer type user that we see often in our audience.

Reply:
@slobentanzer I meant redundant in terms of @laulopezreal 's intention, it does mean something for project packaging and distribution I think and it is a decision that I don't think we should make here right now. As shown above, the produced package manifest of the local editable installation is the same regardless the value of this argument.

Reply:
OK I guess I don't grasp the distinction; I'll leave this to the two of you to sort out. :)

Reply:
Just as a note on the Poetry side of things: if we want to install in non-package-mode, we get a warning in the new Poetry version that will become an error soon. Thus, it would be best practice to include this field sooner rather than later, regardless of which value we choose. Here's the message from the `meta-graph` pipeline:

```
Warning: The current project could not be installed: No file/folder found for package biocypher-meta-graph
If you do not want to install the current project use --no-root.
If you want to use Poetry only for dependency management but not for packaging, you can disable package mode by setting package-mode = false in your pyproject.toml file.
In a future version of Poetry this warning will become an error!
```

While this does not occur if we actually do have a package (as in this case), this is not necessary for a BioCypher pipeline. You can have a pipeline without a package. In all these cases, this error will be relevant.

Reply:
Hello folks :) Thank you so much for looking into this and giving feedback. I am more inclined towards @slobentanzer view of leaving it there. If I were a user, I would understand better the installation procedure if I see that line.
I see both of your points though. The only reason I included it in the first place, is that in my projects, I like to use a `src` folder:
```
my_repository
├───readme.md
│   
├───barconfig
│    bar.xml
├───src
│   ├───my_module1
│   │    python_script.py
│   └───my_module2 (this is not always the case)
```

In this structure, I need to tell poetry where the source code lives so I can import functions or classes between module1 and module2.

Given the structure of this project, this line is not required because it works with the default configuration in poetry. But is good to keep in mind that it is not always the case and you will find projects where that line would be required.

Hope it makes sense!

Reply:
Having said that... this is not a fight that I am going to fight, I understand your view and I believe that it is not incorrect to remove that line. Will change it :) @kpto

Reply:
@slobentanzer @laulopezreal My main concern was that I was not sure what it is doing and I don't like making changes that I don't understand. The poetry documentation is unclear but as it got attention from you all already so I spent some time on experimenting it and here is my findings:

### Overview
1. This is a packaging argument. While local editable installation tries to simulate the end result, it's not always successful and often inaccurate.
2. There is a standard poetry project structure, this argument is redundant if the our structure follows it. The structure is basically having a directory matching the name property in `pytoml` so this PR has already made it matching. Apart from the common `src` structure which requires this argument, this argument is more for namespace fine tuning.

### `from` and `to` argument
Given this project structure, I intentionally make it complicated to show my findings.
```bash
project_dir
├───mymodule_a
│   └───__init__.py
└───src_b
    └───mymodule_b
        └───__init__.py
```
and this argument value:
```python
packages = [
    {include = "mymodule_a", to = "mynamespace"},
    {include = "mymodule_b", from = "src_b"},
]
```
You end up having this structure after `poetry build` and `pip install`
```bash
site-packages
├───mynamespace
│   └───mymodule_a
│       └───__init__.py
└───mymodule_b
    └───__init__.py
```
so you can import these two modules like this, which makes sense to me
```python
import mynamespace.mymodule_a # expected usage
import mymodule_b # expected usage
import mymodule_a # import error which is correct
```

But when installed locally for editing, poetry tries to mimic this but not very successful, it generates two paths:
```
....../project_dir
....../project_dir/src_b
```
Since poetry cannot alter the directory structure of the project, the ultimate behavors becomes:
```python
import mynamespace.mymodule_a # expected usage, but errored because this directory does not exist
import mymodule_b # expected usage
import src_b.mymodule_b # should error but found when searching ....../project_dir
```

### Conclusion
While the change here is not harmful, I still think it should be removed because it's merely repeating the default behavior and it is hard to explain it's intention. Also according to software engineering principles `KISS` and `YAGNI`, such a default behavior overriding argument should be used only when necessary. When creating a new project using `poetry new`, the generated `pytoml` also does not have this argument. We probably will discuss more in depth about how we want to distribute the project and may need it soon though.

@slobentanzer The error you are seeing is because that you project name `biocypher-meta-graph` does not match the module name `meta_graph`. A side note, poetry is smart enough to map `-` to `_` so you don't need to use underscore on project name.

Reply:
Thanks @kpto, that is very useful; I understand this much better now. Agree re the software development principles. If we ever have the need to adjust the default behaviour, we can introduce the parameter at this point. I try to help inexperienced users, but you are correct that this help should not be confusing or beside the point. Maybe we find a way to introduce this in the docs somewhere, so we can at least point the users to a solution or reason.

Reply:
@kpto wow thank you so much for investigating for sharing your findings. I feel that just now I fully understand what poetry does under the hood. Very valuable information as this is something I use recurrently on my projects. 

I've pushed the agreed changes so, hopefully, the PR is okay now. Let me know otherwise :) and thanks again for such an interesting discussion, guys!